File: app/src/main/java/com/erikzuo/portfolioandroidapp/ui/main/work/WorkViewModel.java
package com.erikzuo.portfolioandroidapp.ui.main.work;

import android.arch.lifecycle.MutableLiveData;
import android.databinding.ObservableArrayList;

import com.erikzuo.portfolioandroidapp.data.DataManager;
import com.erikzuo.portfolioandroidapp.data.model.Repo;
import com.erikzuo.portfolioandroidapp.ui.base.BaseViewModel;
import com.erikzuo.portfolioandroidapp.utils.rx.SchedulerProvider;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.functions.Consumer;

/**
 * Created by Soprano on 7/02/2018.
 */

public class WorkViewModel extends BaseViewModel<WorkNavigator> {

    private final ObservableArrayList<Repo> repoList = new ObservableArrayList<>();

    @Inject
    public WorkViewModel(DataManager dataManager, SchedulerProvider schedulerProvider) {
        super(dataManager, schedulerProvider);

        loadRepos();
    }

    public void loadRepos() {
        setIsLoading(true);
        getCompositeDisposable().add(getDataManager()
                .getRepoListApiCall()
                .subscribeOn(getSchedulerProvider().io())
                .observeOn(getSchedulerProvider().ui())
                .subscribe(
                        repos -> {
                            repoList.addAll(repos);
                            setIsLoading(false);
                        }
                )
        );
    }

    public ObservableArrayList<Repo> getRepoList() {
        return repoList;
    }

}

File: app/src/main/java/com/erikzuo/portfolioandroidapp/ui/main/work/WorkViewModel.kt
package com.erikzuo.portfolioandroidapp.ui.main.work

import android.databinding.ObservableArrayList

import com.erikzuo.portfolioandroidapp.data.DataManager
import com.erikzuo.portfolioandroidapp.data.model.Repo
import com.erikzuo.portfolioandroidapp.ui.base.BaseViewModel
import com.erikzuo.portfolioandroidapp.utils.rx.SchedulerProvider

import javax.inject.Inject

/**
 * Created by YifanZuo on 7/02/2018.
 */

class WorkViewModel @Inject
constructor(dataManager: DataManager, schedulerProvider: SchedulerProvider) : BaseViewModel<WorkNavigator>(dataManager, schedulerProvider) {

    val repoList = ObservableArrayList<Repo>()

    init {

        loadRepos()
    }

    private fun loadRepos() {
        setIsLoading(true)
        compositeDisposable.add(dataManager
                .getRepoListApiCall()
                .subscribeOn(schedulerProvider.io())
                .observeOn(schedulerProvider.ui())
                .subscribe { repos ->
                    repoList.addAll(repos)
                    setIsLoading(false)
                }
        )
    }

}
